src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SortDirection.java

```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.ccd.data.casedetails.search;
+
+import java.util.Optional;
+
+public enum SortDirection {
+    ASC, DESC;
+
+    public static SortDirection fromOptionalString(Optional<String> direction) {
```

Comment:
do we really need this method? if you pass a String rather than an Optional you cold leverage the java provided enum method valueOf.
you could deal with the Optional on the caller. Something like: 
metadata.getSortDirection().map(direction -> SortDirection.valueOf(direction)).orElse(SortDirection. DESC)

Reply:
feel free to ignore

Reply:
As far as I understood, the relevant endpoint is accepting the sort direction in a case insensitive manner and moreover, anything other than asc or desc is treated as asc, as per the current functionality.
 
The reason I didn't go with implicit valueOf(String) is because it throws an IllegalArgumentException when the parameter is not a case-sensitively correct element name. If I was to use it, the rest endpoint would probably return an http-500. Now, the custom mapping method provides the case-insensitive mapping functionality as localised and centralised in the most directly relevant code unit, to enable reusability. Having the case-insensivity or any dummy value to ASC mapping inside the query factory or elsewhere would replicate the same business rule in different locations.